src/screens/SignIn/SignIn.tsx

```diff
@@ -0,0 +1,225 @@
+    height: 55,
+    width: "100%",
+    marginBottom: "5%",
+    borderWidth: 1,
+    borderRadius: 5,
+    borderColor: "#4A4B57",
+    padding: "2%",
+  },
+  textInputTitle: {
+    color: "#4A4B57",
+    fontSize: 14,
+  },
+  errorTitle: {
+    color: "#ed0707",
+    fontSize: 14,
+  },
+});
+
+const logo = require("../../assets/bei.jpg");
+
+//  Home Screen Navigation
+function SignInScreen({ navigation }) {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [error, setError] = useState("");
+  const dispatch = useDispatch();
+
+  const isFormValid = () => {
+    if (!/^\w+([.-]?\w+)*@\w+([.-]?\w+)*(\.\w\w+)+$/.test(email)) {
+      return false;
+    }
+
+    if (password.length === 0) {
+      return false;
+    }
+    return true;
+  };
+
+  return (
+    <View style={styles.root}>
+      <View style={{ height: 127 }}>
+        <Image style={styles.image} source={logo} />
+        <Text
+          style={{
+            fontWeight: "bold",
+            fontSize: 20,
+            color: "#4A4B57",
+            paddingLeft: "3%",
+            paddingTop: "2%",
+          }}
+        >
+          Sign In
+        </Text>
+      </View>
+
+      <View
+        style={{
+          flex: 3,
+          paddingVertical: 5,
+          paddingHorizontal: "3%",
+          width: "100%",
+        }}
+      >
+        <ScrollView>
+          <SafeAreaView>
+            <Text style={styles.textInputTitle}>Email</Text>
+            <TextInput
+              placeholder="username@email.com"
+              style={styles.textInput}
+              onChangeText={setEmail}
+              value={email}
+            />
+
+            <Text style={styles.textInputTitle}>Password</Text>
+            <TextInput
+              placeholder="Password"
+              style={styles.textInput}
+              onChangeText={setPassword}
+              value={password}
+              accessibilityHint="d"
+            />
+            <Text style={styles.errorTitle}>{error}</Text>
+          </SafeAreaView>
+        </ScrollView>
+      </View>
+
+      <View
+        style={{
+          flex: 1,
+          alignSelf: "center",
+          paddingHorizontal: "3%",
+          margin: 0,
+        }}
+      >
+        <Button
+          containerStyle={{
+            width: 0.85 * Dimensions.get("window").width,
+            padding: "1%",
+          }}
+          buttonStyle={{
+            backgroundColor: "#005AA3",
+            borderRadius: 4,
+            height: 0.13 * Dimensions.get("window").width,
+          }}
+          titleStyle={styles.buttonTitle}
+          disabled={!isFormValid()}
+          title="Sign In"
+          onPress={() => {
+            setError("");
+            emailSignIn(email, password)
+              .then((res) => {
+                // !! Should add call to backend to retrieve rest of the information !!
+
+                const userObject: AuthUser = {
+                  _id: res.uid,
```

Comment:
This `_id` should be used for MongoDB's id instead of Firebase's

Reply:
Got it. I assumed we would be using the same ids between Mongo and Firebase and Firebase would be the one providing the ids.
Anyway, I changed _id to idToken